src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/ImmutableArrayExtensions.cs

```diff
@@ -121,6 +121,53 @@ public static ImmutableArray<T> WhereAsArray<T>(this ImmutableArray<T> source, F
         return builder.DrainToImmutable();
     }
 
+    /// <summary>
+    /// Returns an <see cref="ImmutableArray{T}"/> that contains no duplicates from the <paramref name="source"/> array
+    /// and returns the most recent copy of each item.
+    /// </summary>
+    public static ImmutableArray<T> GetMostRecentUniqueItems<T>(this ImmutableArray<T> source, IEqualityComparer<T> comparer)
+    {
+        if (source.IsEmpty)
+        {
+            return [];
+        }
+
+#if !NETSTANDARD2_0
+        var uniqueItems = new HashSet<T>(capacity: source.Length, comparer);
+#else
+        var uniqueItems = new HashSet<T>(comparer);
+#endif
+
+        using var stack = new PooledArrayBuilder<T>(capacity: source.Length);
+
+        // Walk the next batch in reverse and to identify unique items.
+        // We push them on a stack so that we can pop them in order later
+        for (var i = source.Length - 1; i >= 0; i--)
```

Comment:
Consider being explicit that `i` is signed integer as that's vital to the loop not being infinite

```suggestion
        for (int i = source.Length - 1; i >= 0; i--)
```

Reply:
I would definitely do that (as it's my preference as well for primitive types). However, the current editorconfig settings in dotnet/razor do not allow that.